nav2_behaviors/src/main.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License. Reserved.
+
+#include <memory>
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_behaviors/spin.hpp"
+#include "nav2_behaviors/back_up.hpp"
+
+int main(int argc, char ** argv)
+{
+  // Force flush of the stdout buffer.
+  setvbuf(stdout, NULL, _IONBF, BUFSIZ);
+
+  rclcpp::init(argc, argv);
+
+  rclcpp::executors::SingleThreadedExecutor exec;
+
+  auto spin = std::make_shared<nav2_behaviors::Spin>();
```

Comment:
Making all behaviors derive from a shared class would allow someone to spin over a pointer vector as another way of architecting this. This also falls in line with the ROS1 implementation of nav_core recoveries.

Reply:
Agree, will push a change shortly.